### PR TITLE
docs: document "and" logic for labels

### PR DESCRIPTION
**Pull request by stevenGravy**
We do not mention that labels have "and" logic.  Also this language was heavy handed in SSH servers only.

### CODE REVIEW

File: docs/pages/access-controls/reference.mdx [hunk 1]
```diff
@@ -186,16 +186,25 @@ spec:
...
   title="Dynamic RBAC"
 >
-  Node labels can be dynamic, i.e. determined at runtime by an output of an executable. In this case, you can implement "permissions follow workload"
+  Resource labels can be dynamic, i.e. determined at runtime by an output of an executable. In this case, you can implement "permissions follow workload"
   policies (eg., any server where PostgreSQL is running becomes *automatically*
   accessible only by the members of the "DBA" group and nobody else).
 </Admonition>
 
-### Extended Node labels syntax
+### Extended Resource labels syntax
```
Comment (zmb3) [inline docs/pages/access-controls/reference.mdx:207]:
```suggestion
### Extende labels syntax
```

**Review by zmb3 — APPROVED** [review]

File: docs/pages/access-controls/reference.mdx [hunk 2]
```diff
@@ -149,10 +149,10 @@ Label              | `v3`, `v4` and `v5` Default   | `v6` Default
 ------------------ | -------------- | ---------------
 `kubernetes_resources` | `[{"kind":"pod", "name":"*", "namespace":"*"}]` | `[]`
 
-## RBAC for hosts
+## RBAC for resources
 
-A Teleport role can also define which hosts (nodes) a user can have access to.
-This works by [labeling nodes](../management/admin/labels.mdx) and listing
+A Teleport role defines which resources (applications, servers, databases,...) a user can have access to.
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:154]:
```suggestion
A Teleport role defines which resources (e.g., applications, servers, and databases) a user can have access to.
```
Fixing ",..."

File: docs/pages/access-controls/reference.mdx [hunk 3]
```diff
@@ -186,16 +186,25 @@ spec:
       'workload': ['database', 'backup']
 ```
 
+Multiple label entries are treated as logical "AND" operations.
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:189]:
```suggestion
Teleport handles multiple label entries with logical "AND" operations.
```
Label entries aren't operations.

File: docs/pages/access-controls/reference.mdx [hunk 4]
```diff
@@ -186,16 +186,25 @@ spec:
...
+As an example this entry would match to databases that have `env: prod` label and
+`region` label `us-west-1` or `eu-central-1`.
```
Comment (ptgott) [inline docs/pages/access-controls/reference.mdx:191]:
```suggestion
As an example this entry would match to databases that have the `env: prod` label and a
`region` label of either `us-west-1` or `eu-central-1`:

```

**Review by ptgott — APPROVED** [review]

**Pull request merged**